src/Github/CreateReleaseTextThroughChangelog.php

```diff
@@ -73,4 +80,36 @@ private function markdownLink(string $text, UriInterface $uri): string
     {
         return '[' . $text . '](' . $uri->__toString() . ')';
     }
+
+    private function normalizeChangelogHeadings(string $changelog): string
+    {
+        $lines         = explode("\n", trim($changelog));
+        $lineCount     = count($lines);
+        $linesToRemove = [];
+        for ($i = 0; $i < $lineCount; $i += 1) {
```

Comment:
This code block is really hard to understand. I suppose most of it is tested well, according to mutation testing results, but it is really making the reader's head spin.

Refactoring can be deferred, since the code is functionally pure, but we need comments about what is being done here.

Reply:
how about using `foreach` instead ? 

```suggestion
        foreach ($lines as $i => $line) {
```

We can eliminate usage of `$lineCount     = count($lines);`